### PR TITLE
Add externs for iron-list

### PR DESCRIPTION
**Pull request by rictic**
Used to get iron-list compiling.

### CODE REVIEW

File: polymer.externs.js [hunk 1]
```diff
@@ -666,10 +739,84 @@ Polymer.Templatizer = {
...
+   *
+   * Example:
+   *
+   *   var model = modelForElement(el);
+   *   if (model.index < 10) {
+   *     model.set('item.checked', true);
+   *   }
+   *
+   * @param {!HTMLElement} el Element for which to return a template model.
+   * @return {Object<PolymerElement>} Model representing the binding scope for
```
Comment (jklein24) [inline polymer.externs.js]: Not sure this type is right here (I just copied it from the docs). Do you know what type this actually is? We should also add either ! or ? to the front of both Object and PolymerElement.

**Comment by jklein24** [comment]
Feels weird reviewing my own code, but wow, nice code!! LGTM :-p

File: polymer.externs.js (hunk 1 above)
Comment (rictic) [inline polymer.externs.js]: Looks like the intent was to say that the returned object implements or extends Polymer.Base. In our type system I think that most closely corresponds to PolymerElement.

Here's the source to double-check this: https://github.com/Polymer/polymer/blob/8868fdc9f231c615e554ec21833efe0d62de4718/src/lib/template/templatizer.html#L411

Comment (jklein24) [inline polymer.externs.js]: Interesting... I think you're right. It does seem weird that "model" is an instance of PolymerElement, though. Shrug. Go with {?PolymerElement}

**Comment by rictic** [comment]
As discussed, refined type for modelForElement's return. Merging!

**Pull request merged**